subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -166,6 +166,8 @@ public void validateMutation(MutationType type) {
     private boolean dependenciesModified;
     private boolean canBeConsumed = true;
     private boolean canBeResolved = true;
+
+    private boolean canBeMutated = true;
```

Comment:
I was referring to the fact that all other mutations are checked differently: In the `validateMutation` method, using the `MutationType` enum and using the `resolveState` to tell whether the mutation should be allowed.

This change introduces a new field instead of adding to the existing list of mutation types. Not a big deal, I just thought that maybe we could unify this.